src/main/java/edu/uw/edm/edmzuulproxy/certificateauthorizer/service/impl/CertificateAuthorizerServiceImpl.java

```diff
@@ -31,63 +33,64 @@
+        for (CompiledCertificateAuthorization certificateAuthorizationDAO : byCertificateName) {
+            if (isAuthorizedForRequest(certificateAuthorizationDAO, httpMethod, uri, user)) {
                 return true;
             }
         }
 
         return false;
     }
 
-    private boolean isAuthorizedForRequest(CertificateAuthorization certificateAuthorization, HttpMethod httpMethod, String uri, User user) {
+    private boolean isAuthorizedForRequest(CompiledCertificateAuthorization certificateAuthorizationDAO, HttpMethod httpMethod, String uri, User user) {
 
-        final boolean methodIsAuthorized = methodIsAuthorized(httpMethod, certificateAuthorization);
-        final boolean uriIsAuthorized = uriIsAuthorized(uri, certificateAuthorization);
-        final boolean uwGroupsMatch = uwGroupsMatch(user, certificateAuthorization);
+        final boolean authorized = methodIsAuthorized(httpMethod, certificateAuthorizationDAO) &&
+                uriIsAuthorized(uri, certificateAuthorizationDAO)
+                && uwGroupsMatch(user, certificateAuthorizationDAO);
 
-
-        final boolean authorized = methodIsAuthorized &&
-                uriIsAuthorized &&
-                uwGroupsMatch;
-
-        log.trace("method : {} , uri : {}, uwGroups: {} => result : {}", methodIsAuthorized, uriIsAuthorized, uwGroupsMatch, authorized);
         return authorized;
 
     }
 
-    private boolean uwGroupsMatch(User user, CertificateAuthorization certificateAuthorization) {
-        return certificateAuthorization.getUwGroups() == null || certificateAuthorization.getUwGroups().contains(UW_GROUPS_WILDCARD) || userIsMemberOfOneGroup(user, certificateAuthorization);
+    private boolean uwGroupsMatch(User user, CompiledCertificateAuthorization certificateAuthorization) {
+        return certificateAuthorization.getUwGroups() == null
+                || certificateAuthorization.getUwGroups().contains(UW_GROUPS_WILDCARD)
+                || userIsMemberOfOneGroup(user, certificateAuthorization);
     }
 
-    private boolean userIsMemberOfOneGroup(User user, CertificateAuthorization certificateAuthorization) {
-        //TODO
-        return false;
-    }
+    private boolean userIsMemberOfOneGroup(User user, CompiledCertificateAuthorization certificateAuthorization) {
+
 
-    private boolean uriIsAuthorized(String uri, CertificateAuthorization certificateAuthorization) {
-        final Pattern pattern = Pattern.compile(certificateAuthorization.getUriRegex());
+        return user!=null && user.getAuthorities()
+                .stream()
+                .map(GrantedAuthority::getAuthority)
+                .anyMatch(authority -> certificateAuthorization.getUwGroups().contains(authority));
+    }
 
-        return pattern.matcher(uri).matches();
+    private boolean uriIsAuthorized(String uri, CompiledCertificateAuthorization certificateAuthorization) {
+        return certificateAuthorization.getUriRegex().matcher(uri).matches();
```

Comment:
should we check that getUriRegex() is not null?

Reply:
it'd fail before in `edu.uw.edm.edmzuulproxy.certificateauthorizer.service.impl.CertificateAuthorizationRetriever#toCompiledCertificateAuthorization`